### PR TITLE
added tabs component

### PR DESCRIPTION
**Pull request by kaiwerther**
added tabs component based on figma design

### CODE REVIEW

File: src/components/ifxTabs/ifxTabs.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  const tabsElement = document.createElement('ifx-tabs') as HTMLIfxTabsElement;
+  tabsElement.setAttribute('orientation', args.orientation);
+  if (args.small) {
+    tabsElement.setAttribute('small', ''); // Add this line
+  }
+  const tabsArray = Array.from({ length: args.amountOfTabs }, (_, i) => `Tab ${i + 1}`);
+  tabsElement.tabs = tabsArray;
+  for (let i = 0; i < args.amountOfTabs; i++) {
+    const tabContent = document.createElement('div');
+    tabContent.slot = `tab-content-${i}`;
```
Comment (tishoyanchev) [inline src/components/ifxTabs/ifxTabs.stories.tsx:21]: Why not rename `tab-content-1` to `tab-1`?

File: src/components/ifxTabs/ifxTabs.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
+export default {
+  title: 'Components/Tabs',
+  component: 'ifx-tabs',
+  argTypes: {
+    amountOfTabs: { control: 'number' },
```
Comment (tishoyanchev) [inline src/components/ifxTabs/ifxTabs.stories.tsx:5]: When `amountOfTabs` is `0`, there is an error in the console. 
Why not rename `amountOfTabs` to `tabs`. The word 'amount' is not correct here, and there should be space between the words.

File: src/components/ifxTabs/ifxTabs.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        borderElement.style.top = '';
+        borderElement.style.height = '';
+      } else {
+        borderElement.style.top = `${this.tabRefs[index].offsetTop}px`;
+        borderElement.style.height = `${this.tabRefs[index].offsetHeight}px`;
+        borderElement.style.left = '';
+        borderElement.style.width = '';
+      }
+    }
+  }
```
Comment (tishoyanchev) [inline src/components/ifxTabs/ifxTabs.tsx:39]: Why not replace the code inside the if condition with a `State` that gets updated, which then conditionally adds a class on the element, and these styles are all defined in the `scss file`? That would save a lot of code, make it consistent and easier to read and edit.

File: src/components/ifxTabs/ifxTabs.scss [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+.tab-item {
+  padding: 0.5rem 1rem ;
+  cursor: pointer;
+  position: relative;
+}
+
+.tabs.vertical .tab-item {
+  border-right: 2px solid transparent;
+  min-width: 7em;
```
Comment (tishoyanchev) [inline src/components/ifxTabs/ifxTabs.scss:59]: `min-width` of `7em` seems too large. The space between the tab and the content is larger than in the horizontal variant.

File: src/components/ifxTabs/ifxTabs.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  return tabsElement;
+};
+
+export const Default = Template.bind({});
+Default.args = {
+  amountOfTabs: 3,
+  orientation: 'horizontal',
+};
+
+export const Small = Template.bind({});
```
Comment (tishoyanchev) [inline src/components/ifxTabs/ifxTabs.stories.tsx:41]: Why is there a need for a `small variant`, when there already is a control option for `small text`?

**Review by tishoyanchev — CHANGES_REQUESTED** [review]

File: src/components/ifxTabs/ifxTabs.tsx [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+export type TabOrientation = 'horizontal' | 'vertical';
+
+
+@Component({
+  tag: 'ifx-tabs',
+  styleUrl: 'ifxTabs.scss',
+  shadow: true
+})
+
+export class IfxTabs {
```
[the remark this answers is not in the document]
Comment (kaiwerther) [reply src/components/ifxTabs/ifxTabs.tsx:13]: discussion in other PR. I'll close this here

File: src/components/ifxTabs/ifxTabs.tsx [hunk 7]
```diff
@@ -0,0 +1,70 @@
+//ifxTabs.tsx
+import { Component, h, Prop, State, Element } from '@stencil/core';
+
+export type TabOrientation = 'horizontal' | 'vertical';
```
[the remark this answers is not in the document]
Comment (kaiwerther) [reply src/components/ifxTabs/ifxTabs.tsx:4]: With your solution the user can (even in a typescript environment) have orientation set to "abc" for example.

File: src/components/ifxTabs/ifxTabs.tsx [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+            >
+              {tab}
+            </li>
+          ))}
+          <div class="active-border"></div>
+        </ul>
+        <div class="tab-content">
+          {this.tabs.map((_, index) => (
+            <div style={{ display: index === this.activeTabIndex ? 'block' : 'none' }}>
+              <slot name={`tab-content-${index}`} />
```
[the remark this answers is not in the document]
Comment (kaiwerther) [reply src/components/ifxTabs/ifxTabs.tsx:63]: yes - without tabs no slots for the tabs? If you have no tabs why do you expect the slot to be there?

File: src/components/ifxTabs/ifxTabs.tsx [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+
+  componentDidLoad() {
+    this.setActiveTab(this.activeTabIndex);
+  }
+
+  render() {
+    return (
+      <div class={`tabs ${this.orientation} ${this.small ? 'small' : ''}`}>
+        <ul class="tabs-list">
+          {this.tabs.map((tab, index) => (
```
[the remark this answers is not in the document]
Comment (kaiwerther) [reply src/components/ifxTabs/ifxTabs.tsx:49]: tabs should be filled from the user. Why do you think this is only used in storybook? Storybook is an example in how it should be used by the user.

File: src/components/ifxTabs/ifxTabs.stories.tsx [hunk 10]
```diff
@@ -0,0 +1,46 @@
+export default {
+  title: 'Components/Tabs',
+  component: 'ifx-tabs',
+  argTypes: {
+    amountOfTabs: { control: 'number' },
+    orientation: { control: { type: 'select', options: ['horizontal', 'vertical'] } },
+    small: { control: 'boolean' }, // Add this line
```
[the remark this answers is not in the document]
Comment (kaiwerther) [reply src/components/ifxTabs/ifxTabs.stories.tsx:7]: Is this an issue? We use the boolean control from storybook for an boolean field.

File: src/components/ifxTabs/ifxTabs.stories.tsx [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+  },
+};
+
+const Template = (args) => {
+  const tabsElement = document.createElement('ifx-tabs') as HTMLIfxTabsElement;
+  tabsElement.setAttribute('orientation', args.orientation);
+  if (args.small) {
+    tabsElement.setAttribute('small', ''); // Add this line
+  }
+  const tabsArray = Array.from({ length: args.amountOfTabs }, (_, i) => `Tab ${i + 1}`);
```
[the remark this answers is not in the document]
Comment (kaiwerther) [reply src/components/ifxTabs/ifxTabs.stories.tsx:17]: yes. slot names need to match the amount of tabs.
Shit in shit out ;)
If you have 2 slots and try to assign the content of tab #10 which doesn't exist I would expect an error

File: src/components/ifxTabs/ifxTabs.tsx [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+              onClick={() => this.setActiveTab(index)}
+            >
+              {tab}
+            </li>
+          ))}
+          <div class="active-border"></div>
+        </ul>
+        <div class="tab-content">
+          {this.tabs.map((_, index) => (
+            <div style={{ display: index === this.activeTabIndex ? 'block' : 'none' }}>
```
[the remark this answers is not in the document]
Comment (kaiwerther) [reply src/components/ifxTabs/ifxTabs.tsx:62]: I would aggre that inline styles are bad practice but for it's good enough for setting visibility.

File: src/components/ifxTabs/ifxTabs.stories.tsx (hunk 2 above)
Comment (kaiwerther) [reply src/components/ifxTabs/ifxTabs.stories.tsx:5]: Fixed the console.log error
Added names. We should propably add this to every story.

File: src/components/ifxTabs/ifxTabs.scss (hunk 4 above)
Comment (kaiwerther) [reply src/components/ifxTabs/ifxTabs.scss:59]: Let's overhaul all the spacing when we have to tokens for it.

File: src/components/ifxTabs/ifxTabs.stories.tsx (hunk 5 above)
Comment (kaiwerther) [reply src/components/ifxTabs/ifxTabs.stories.tsx:41]: There is one in Figma - it's not just the text - the spacing is a little bit different.

**Review by tishoyanchev — CHANGES_REQUESTED** [review]

File: src/components/ifxTabs/ifxTabs.tsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (verena-ifx) [reply src/components/ifxTabs/ifxTabs.tsx:53]: The formatting when I simply copy/paste the code from the story, is off. I only get the content of the tabs, but not actual tabs.

**Pull request merged**